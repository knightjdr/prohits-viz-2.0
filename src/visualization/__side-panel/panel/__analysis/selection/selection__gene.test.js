import React from 'react';
import { shallow } from 'enzyme';

import Gene from './selection__gene';

const arrangeSelected = jest.fn();
const listSwap = jest.fn();

beforeEach(() => {
  /* Clear call count */
  arrangeSelected.mockClear();
  listSwap.mockClear();
});

describe('Gene select', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(
      <Gene
        arrangeSelected={arrangeSelected}
        columnRef={{}}
        columns={['a', 'b']}
        columnsSelected={['c']}
        listSelect={jest.fn()}
        listSwap={listSwap}
        openContextMenu={jest.fn()}
        rowRef={{}}
        rows={['e']}
        rowsSelected={['d', 'f']}
      />,
    );
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should swap columns with columnsSelected on button click', () => {
    wrapper.find('button').at(0).simulate('click');
    expect(listSwap).toHaveBeenCalledWith('columns', 'columnsSelected');
  });

  it('should swap columnsSelected with columns on button click', () => {
    wrapper.find('button').at(1).simulate('click');
    expect(listSwap).toHaveBeenCalledWith('columnsSelected', 'columns', 'columnMap');
  });

  it('should call arrange selected for columns with decrement on button click', () => {
    wrapper.find('button').at(2).simulate('click');
    expect(arrangeSelected).toHaveBeenCalledWith('columnsSelected', -1);
  });

  it('should call arrange selected for columns with increment on button click', () => {
    wrapper.find('button').at(3).simulate('click');
    expect(arrangeSelected).toHaveBeenCalledWith('columnsSelected', 1);
  });

  it('should swap rows with rowsSelected on button click', () => {
    wrapper.find('button').at(4).simulate('click');
    expect(listSwap).toHaveBeenCalledWith('rows', 'rowsSelected');
  });

  it('should swap rowsSelected with rows on button click', () => {
    wrapper.find('button').at(5).simulate('click');
    expect(listSwap).toHaveBeenCalledWith('rowsSelected', 'rows', 'rowMap');
  });

  it('should call arrange selected for rows with decrement on button click', () => {
    wrapper.find('button').at(6).simulate('click');
    expect(arrangeSelected).toHaveBeenCalledWith('rowsSelected', -1);
  });

  it('should call arrange selected for rows with increment on button click', () => {
    wrapper.find('button').at(7).simulate('click');
    expect(arrangeSelected).toHaveBeenCalledWith('rowsSelected', 1);
  });
});
