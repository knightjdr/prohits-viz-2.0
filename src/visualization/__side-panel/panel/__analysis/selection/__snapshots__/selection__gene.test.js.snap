// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gene select should render 1`] = `
<div
  className="selection"
>
  <div
    className="selection__grid"
  >
    <div
      className="selection__header"
    >
      Columns
    </div>
    <div />
    <div
      className="selection__header"
    >
      Selected
    </div>
    <div />
    <Select
      canPaste={false}
      listSelect={[Function]}
      openContextMenu={[Function]}
      options={
        Array [
          "a",
          "b",
        ]
      }
      setRef={Object {}}
      target="columns"
    />
    <div
      className="selection__button-group"
    >
      <button
        className="selection__select-arrow"
        onClick={[Function]}
        type="button"
      >
        <FontAwesomeIcon$1
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f062",
                "M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z",
              ],
              "iconName": "arrow-up",
              "prefix": "fas",
            }
          }
          listItem={false}
          mask={null}
          name=""
          pull={null}
          pulse={false}
          rotation={90}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </button>
      <button
        className="selection__select-arrow"
        onClick={[Function]}
        type="button"
      >
        <FontAwesomeIcon$1
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f062",
                "M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z",
              ],
              "iconName": "arrow-up",
              "prefix": "fas",
            }
          }
          listItem={false}
          mask={null}
          name=""
          pull={null}
          pulse={false}
          rotation={270}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </button>
    </div>
    <Select
      canPaste={true}
      listSelect={[Function]}
      openContextMenu={[Function]}
      options={
        Array [
          "c",
        ]
      }
      setRef={null}
      target="columnsSelected"
    />
    <div
      className="selection__button-group"
    >
      <button
        className="selection__select-arrow"
        onClick={[Function]}
        type="button"
      >
        <FontAwesomeIcon$1
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f062",
                "M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z",
              ],
              "iconName": "arrow-up",
              "prefix": "fas",
            }
          }
          listItem={false}
          mask={null}
          name=""
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </button>
      <button
        className="selection__select-arrow"
        onClick={[Function]}
        type="button"
      >
        <FontAwesomeIcon$1
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f062",
                "M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z",
              ],
              "iconName": "arrow-up",
              "prefix": "fas",
            }
          }
          listItem={false}
          mask={null}
          name=""
          pull={null}
          pulse={false}
          rotation={180}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </button>
    </div>
  </div>
  <div
    className="selection__grid"
  >
    <div
      className="selection__header"
    >
      Rows
    </div>
    <div />
    <div
      className="selection__header"
    >
      Selected
    </div>
    <div />
    <Select
      canPaste={false}
      listSelect={[Function]}
      openContextMenu={[Function]}
      options={
        Array [
          "e",
        ]
      }
      setRef={Object {}}
      target="rows"
    />
    <div
      className="selection__button-group"
    >
      <button
        className="selection__select-arrow"
        onClick={[Function]}
        type="button"
      >
        <FontAwesomeIcon$1
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f062",
                "M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z",
              ],
              "iconName": "arrow-up",
              "prefix": "fas",
            }
          }
          listItem={false}
          mask={null}
          name=""
          pull={null}
          pulse={false}
          rotation={90}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </button>
      <button
        className="selection__select-arrow"
        onClick={[Function]}
        type="button"
      >
        <FontAwesomeIcon$1
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f062",
                "M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z",
              ],
              "iconName": "arrow-up",
              "prefix": "fas",
            }
          }
          listItem={false}
          mask={null}
          name=""
          pull={null}
          pulse={false}
          rotation={270}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </button>
    </div>
    <Select
      canPaste={true}
      listSelect={[Function]}
      openContextMenu={[Function]}
      options={
        Array [
          "d",
          "f",
        ]
      }
      setRef={null}
      target="rowsSelected"
    />
    <div
      className="selection__button-group"
    >
      <button
        className="selection__select-arrow"
        onClick={[Function]}
        type="button"
      >
        <FontAwesomeIcon$1
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f062",
                "M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z",
              ],
              "iconName": "arrow-up",
              "prefix": "fas",
            }
          }
          listItem={false}
          mask={null}
          name=""
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </button>
      <button
        className="selection__select-arrow"
        onClick={[Function]}
        type="button"
      >
        <FontAwesomeIcon$1
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f062",
                "M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z",
              ],
              "iconName": "arrow-up",
              "prefix": "fas",
            }
          }
          listItem={false}
          mask={null}
          name=""
          pull={null}
          pulse={false}
          rotation={180}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </button>
    </div>
  </div>
</div>
`;
